src/main/scala/scala/tools/partest/nest/Runner.scala

```diff
@@ -759,7 +759,7 @@ class SuiteRunner(
     // |Java Classpath:             ${sys.props("java.class.path")}
   }
 
-  def onFinishTest(testFile: File, result: TestState): TestState = result
+  def onFinishTest(@deprecated("unused","") testFile: File, result: TestState): TestState = result
```

Comment:
this is fighting a symptom..

Reply:
Shouldn't it only warn on effectively final methods?

Reply:
@dwijnand I think that weakens the warning. Maybe two modes? strict vs non-strict? By the time you've written `final`, you're less likely to have made a careless mistake such as this is designed to catch. So `strict` mode would mean "I think I was careful."

Reply:
Sure, it weakens it, but I think I prefer a warning that you don't have to workaround.

(As a complete aside, IMO things should be final by default and you should have to annotate it `open`, or something, rather than have to declare things `final`).